### PR TITLE
feat: Next.js bootstrapping

### PR DESCRIPTION
**Pull request by aomi**
# Description
This PR bootstraps Next.js within the repository. It actually doesn't get used and all the commands that work for Next.js are documented in the README.

This PR also updates some core dependencies which have minor API changes of which have been properly addressed. The PR should not change the behaviour of the application and any such changes are errors/bugs introduced in this PR.

Subsequent work will start using Next.js and configure the deployment for it etc.

## Checklist

- [x] The code follows all style guidelines.
- [x] The code passes all required tests.
- [x] The code is documented.
- [x] The code includes tests.
- [ ] I have self-reviewed my changes and have done QA.

### CODE REVIEW

**Comment by aomi** [comment]
@szeckirjr added you explicitly as a reviewer as I had to update some props within the shared timetable code and want to check with you about it.

File: src/pages/import/index.tsx [hunk 1]
```diff
@@ -87,7 +87,7 @@ export function ImportTimetable(): JSX.Element {
           {!loading && data ? calendarView ? calendarComponent : listView : <Spinner size="xl" />}
         </VStack>
       ) : (
-        <VStack pt={2} w="100%" height="100%" overflow="hidden" grow={1}>
+        <VStack pt={2} w="100%" height="100%" overflow="hidden" flexGrow={1}>
```
Comment (keithradford) [inline src/pages/import/index.tsx:90]: is there a reason for prefixing these props with `flex`?

Comment (keithradford) [reply src/pages/import/index.tsx:90]: i thought chakra added these  "sub-`flex`" props to flex components to simplify the prop names

Comment (aomi) [reply src/pages/import/index.tsx:90]: Updating removed those props so I had to change them.

**Review by keithradford — APPROVED** [review]
I know we aren't using them yet but is it expected most the `next` npm commands fail?

**Pull request merged**